workitem/workitemtype_repository_blackbox_test.go

```diff
@@ -30,118 +28,120 @@ func (s *workItemTypeRepoBlackBoxTest) SetupTest() {
 	workitem.ClearGlobalWorkItemTypeCache()
 }
 
-func (s *workItemTypeRepoBlackBoxTest) TestCreateLoadWIT() {
-
-	wit, err := s.repo.Create(s.Ctx, space.SystemSpace, nil, nil, "foo_bar", nil, "fa-bomb", map[string]workitem.FieldDefinition{
-		"foo": {
-			Required: true,
-			Type:     &workitem.SimpleType{Kind: workitem.KindFloat},
-		},
+func (s *workItemTypeRepoBlackBoxTest) TestExists() {
+	s.T().Run("wit exists", func(t *testing.T) {
+		// given
+		fxt := tf.NewTestFixture(t, s.DB, tf.WorkItemTypes(1))
+		// when
+		err := s.repo.CheckExists(s.Ctx, fxt.WorkItemTypes[0].ID.String())
+		// then
+		require.Nil(s.T(), err)
 	})
-	require.Nil(s.T(), err)
-	require.NotNil(s.T(), wit)
-	require.NotNil(s.T(), wit.ID)
 
-	// Test that we can create a WIT with the same name as before.
-	wit3, err := s.repo.Create(s.Ctx, space.SystemSpace, nil, nil, "foo_bar", nil, "fa-bomb", map[string]workitem.FieldDefinition{})
-	require.Nil(s.T(), err)
-	require.NotNil(s.T(), wit3)
-	require.NotNil(s.T(), wit3.ID)
-
-	wit2, err := s.repo.Load(s.Ctx, space.SystemSpace, wit.ID)
-	require.Nil(s.T(), err)
-	require.NotNil(s.T(), wit2)
-	require.NotNil(s.T(), wit2.Fields)
-	field := wit2.Fields["foo"]
-	require.NotNil(s.T(), field)
-	assert.Equal(s.T(), workitem.KindFloat, field.Type.GetKind())
-	assert.Equal(s.T(), true, field.Required)
+	s.T().Run("wit doesn't exist", func(t *testing.T) {
+		// given
+		nonExistingWorkItemTypeID := uuid.NewV4()
+		// when
+		err := s.repo.CheckExists(s.Ctx, nonExistingWorkItemTypeID.String())
+		// then
+		require.IsType(t, errors.NotFoundError{}, err)
+	})
 }
+func (s *workItemTypeRepoBlackBoxTest) TestCreate() {
+	s.T().Run("create and load", func(t *testing.T) {
+		// Test that we can create two WITs with the same name, the second has a
+		// different fields definition
+		fxt := tf.NewTestFixture(t, s.DB,
+			tf.WorkItemTypes(2,
+				tf.SetWorkItemTypeNames([]string{"foo_bar", "foo_bar"}),
+				func(fxt *tf.TestFixture, idx int) error {
+					if idx == 1 {
+						fxt.WorkItemTypes[1].Fields = map[string]workitem.FieldDefinition{
+							"foo": {
+								Required: true,
+								Type:     &workitem.SimpleType{Kind: workitem.KindFloat},
+							},
+						}
+					}
+					return nil
+				},
+			),
+		)
+		require.Equal(t, "foo_bar", fxt.WorkItemTypes[0].Name)
+		require.Equal(t, "foo_bar", fxt.WorkItemTypes[1].Name)
+
+		require.NotNil(t, fxt.WorkItemTypes[1])
+		require.NotNil(t, fxt.WorkItemTypes[1].Fields)
```

Comment:
How about adding this one more assert?
`assert.NotEqual(t, fxt.WorkItemTypes[1].Fields, fxt.WorkItemTypes[0].Fields)`

Reply:
I think this might be problematic because the fields contain time values also and comparing them won't work as we've seen earlier.